### ISSUE TITLE
training loss

### ISSUE THREAD

**Issue by LeoCHANGcn** [eriklindernoren/PyTorch-YOLOv3#81]
I got the following outputs. It seems the model learns nothing! And why the loss_w and loss_h are so large? And it also seems difficult to minimize 3 losses at 3 different scale.
```shell
************epoch: 0 ***************
total loss: 348640.53125
loss1:913.8490600585938, loss2: 23764.802734375, loss3: 323961.875
x loss: 0.3195413450400035
y loss: 0.3137897203365962
w loss: 66638.80253092448
h loss: 49531.71296183268
conf loss: 41.787367502848305
cls loss: 0.5715278486410776
recall: 0.0
precision: 0.0
************epoch: 1 ***************
total loss: 46213.30859375
loss1:957.5968627929688, loss2: 8531.0283203125, loss3: 36724.68359375
x loss: 0.32575063904126483
y loss: 0.3347550928592682
w loss: 9378.899495442709
h loss: 5994.662760416667
conf loss: 29.653959274291992
cls loss: 0.5596082905928293
recall: 0.0
precision: 0.0
************epoch: 2 ***************
total loss: 19012.55078125
loss1:585.091552734375, loss2: 4879.197265625, loss3: 13548.2626953125
x loss: 0.27623510360717773
y loss: 0.3009087145328522
w loss: 2949.1616414388022
h loss: 3356.6504516601562
conf loss: 30.56474431355794
cls loss: 0.5631969173749288
recall: 0.0
precision: 0.0
************epoch: 3 ***************
total loss: 23540.45703125
loss1:774.5172729492188, loss2: 4625.90576171875, loss3: 18140.03515625
x loss: 0.37653597195943195
y loss: 0.35577158133188885
w loss: 2872.778523763021
h loss: 4946.129964192708
conf loss: 26.616029103597004
cls loss: 0.5623655120531718
recall: 0.0
precision: 0.0
************epoch: 4 ***************
total loss: 20757.111328125
loss1:766.176025390625, loss2: 4598.87255859375, loss3: 15392.0625
x loss: 0.360775629679362
y loss: 0.3116363187630971
w loss: 4197.816589355469
h loss: 2693.129648844401
conf loss: 26.844149271647137
cls loss: 0.5738274653752645
recall: 0.0
precision: 0.0
************epoch: 5 ***************
total loss: 25133.517578125
loss1:840.18701171875, loss2: 3553.506591796875, loss3: 20739.82421875
x loss: 0.42523420850435895
y loss: 0.31204745173454285
w loss: 6128.641143798828
h loss: 2216.5249938964844
conf loss: 31.396429697672527
cls loss: 0.5392687817414602
recall: 0.0
precision: 0.0
************epoch: 6 ***************
total loss: 9306.60546875
loss1:701.0897216796875, loss2: 3326.40771484375, loss3: 5279.1083984375
x loss: 0.34626181920369464
y loss: 0.36598289012908936
w loss: 1723.748306274414
h loss: 1347.4306945800781
conf loss: 29.737626393636067
cls loss: 0.5730269153912863
recall: 0.0
precision: 0.0
************epoch: 7 ***************
total loss: 6939.2958984375
loss1:320.3407287597656, loss2: 2464.346923828125, loss3: 4154.60791015625
x loss: 0.31976256767908734
y loss: 0.3466716210047404
w loss: 1135.3634287516277
h loss: 1155.8743184407551
conf loss: 20.62406349182129
cls loss: 0.5704464316368103
recall: 0.0
precision: 0.0
************epoch: 8 ***************
total loss: 7135.021484375
loss1:287.25054931640625, loss2: 2073.951416015625, loss3: 4773.8193359375
x loss: 0.3412250876426697
y loss: 0.2804672618707021
w loss: 1193.9500122070312
h loss: 1160.0985946655273
conf loss: 23.113460540771484
cls loss: 0.5565606753031412
recall: 0.0
precision: 0.0
************epoch: 9 ***************
total loss: 8677.26953125
loss1:330.572265625, loss2: 3345.734375, loss3: 5000.96240234375
x loss: 0.32532837986946106
y loss: 0.317234605550766
w loss: 1415.9094340006511
h loss: 1453.3638865152996
conf loss: 21.933124542236328
cls loss: 0.5739687283833822
recall: 0.0
precision: 0.0
Save Model at ./models/yolo/10.pth

************epoch: 10 ***************
total loss: 6418.7900390625
loss1:274.5879821777344, loss2: 1392.53466796875, loss3: 4751.66748046875
x loss: 0.28437716762224835
y loss: 0.33507784207661945
w loss: 1440.7245330810547
h loss: 677.5979334513346
conf loss: 20.091039657592773
cls loss: 0.5636382301648458
recall: 0.0
precision: 0.0
************epoch: 11 ***************
total loss: 5476.48095703125
loss1:213.6429443359375, loss2: 2110.10302734375, loss3: 3152.73486328125
x loss: 0.36500510573387146
y loss: 0.33946532011032104
w loss: 1032.5993677775066
h loss: 772.3858515421549
conf loss: 19.236771901448567
cls loss: 0.5671559770901998
recall: 0.0
precision: 0.0
************epoch: 12 ***************
total loss: 6606.1416015625
loss1:448.2239074707031, loss2: 2842.820068359375, loss3: 3315.097412109375
x loss: 0.38653305172920227
y loss: 0.42898163199424744
w loss: 1399.6859334309895
h loss: 780.3089497884115
conf loss: 20.688037872314453
cls loss: 0.5487240254878998
recall: 0.0
precision: 0.0
************epoch: 13 ***************
total loss: 4365.15234375
loss1:248.04583740234375, loss2: 1787.0714111328125, loss3: 2330.03515625
x loss: 0.32281092802683514
y loss: 0.29365310072898865
w loss: 763.6775690714518
h loss: 668.6572901407877
conf loss: 21.535467783610027
cls loss: 0.5639757513999939
recall: 0.0
precision: 0.0
************epoch: 14 ***************
total loss: 5318.37109375
loss1:485.59857177734375, loss2: 1417.810791015625, loss3: 3414.9619140625
x loss: 0.32043687999248505
y loss: 0.3296237587928772
w loss: 789.0275166829427
h loss: 962.8980611165365
conf loss: 19.65449333190918
cls loss: 0.5602083007494608
recall: 0.0
precision: 0.0
************epoch: 15 ***************
total loss: 7685.54736328125
loss1:553.9857177734375, loss2: 2559.395263671875, loss3: 4572.16650390625
x loss: 0.3545884887377421
y loss: 0.27861021955808
w loss: 943.6536153157552
h loss: 1598.8468424479167
conf loss: 18.156211217244465
cls loss: 0.5594308376312256
recall: 0.0
precision: 0.0
************epoch: 16 ***************
total loss: 4582.3203125
loss1:223.67442321777344, loss2: 1957.3697509765625, loss3: 2401.27587890625
x loss: 0.32813094059626263
y loss: 0.3751894036928813
w loss: 863.9996821085612
h loss: 645.4363428751627
conf loss: 16.735541661580402
cls loss: 0.5651102463404337
recall: 0.0
precision: 0.0
************epoch: 17 ***************
total loss: 4119.8759765625
loss1:232.18629455566406, loss2: 1401.507080078125, loss3: 2486.182373046875
x loss: 0.3908514479796092
y loss: 0.3582785526911418
w loss: 630.0600433349609
h loss: 725.3446706136068
conf loss: 16.57646719614665
cls loss: 0.561546266078949
recall: 0.0
precision: 0.0
************epoch: 18 ***************
total loss: 3505.315673828125
loss1:207.05886840820312, loss2: 1186.09423828125, loss3: 2112.16259765625
x loss: 0.35865628719329834
y loss: 0.28687164187431335
w loss: 549.4932861328125
h loss: 599.9891103108724
conf loss: 17.738918940226238
cls loss: 0.571782390276591
recall: 0.0
precision: 0.0
************epoch: 19 ***************
total loss: 4809.5947265625
loss1:205.74758911132812, loss2: 1846.8729248046875, loss3: 2756.97412109375
x loss: 0.2718834529320399
y loss: 0.33982659379641217
w loss: 884.3853632609049
h loss: 699.2422027587891
conf loss: 18.386398633321125
cls loss: 0.5724744002024332
recall: 0.0
precision: 0.0
Save Model at ./models/yolo/20.pth

************epoch: 20 ***************
total loss: 4410.7119140625
loss1:322.09478759765625, loss2: 1302.790283203125, loss3: 2785.8271484375
x loss: 0.312864288687706
y loss: 0.38178516427675885
w loss: 447.5242055257161
h loss: 1003.9290669759115
conf loss: 17.54058265686035
cls loss: 0.5489791532357534
recall: 0.0
precision: 0.0
************epoch: 21 ***************
total loss: 4315.7734375
loss1:342.5912780761719, loss2: 1860.9486083984375, loss3: 2112.233642578125
x loss: 0.2975549300511678
y loss: 0.30126820007960003
w loss: 625.9556732177734
h loss: 793.6216023763021
conf loss: 17.84151490529378
cls loss: 0.5735308130582174
recall: 0.0
precision: 0.0
************epoch: 22 ***************
total loss: 6517.6533203125
loss1:431.5964050292969, loss2: 2145.180908203125, loss3: 3940.876220703125
x loss: 0.337896207968394
y loss: 0.35937540729840595
w loss: 1358.1378428141277
h loss: 792.8054351806641
conf loss: 20.36822001139323
cls loss: 0.5423655311266581
recall: 0.0
precision: 0.0
************epoch: 23 ***************
total loss: 3638.186767578125
loss1:188.5393829345703, loss2: 1858.5115966796875, loss3: 1591.1357421875
x loss: 0.3132857879002889
y loss: 0.36100200315316516
w loss: 804.5006052652994
h loss: 390.0103251139323
conf loss: 16.977762540181477
cls loss: 0.5659379363059998
recall: 0.0
precision: 0.0
************epoch: 24 ***************
total loss: 6789.03125
loss1:315.0848083496094, loss2: 2686.7861328125, loss3: 3787.160400390625
x loss: 0.2969197730223338
y loss: 0.3670845031738281
w loss: 996.2614669799805
h loss: 1248.6476389567058
conf loss: 16.862656593322754
cls loss: 0.5746623873710632
recall: 0.0
precision: 0.0
************epoch: 25 ***************
total loss: 3569.08251953125
loss1:207.31924438476562, loss2: 1765.5067138671875, loss3: 1596.256591796875
x loss: 0.315743828813235
y loss: 0.3395520746707916
w loss: 612.9581604003906
h loss: 559.0120112101237
conf loss: 16.493522961934406
cls loss: 0.5751583377520243
recall: 0.0
precision: 0.0
************epoch: 26 ***************
total loss: 6508.1845703125
loss1:498.76519775390625, loss2: 2710.870849609375, loss3: 3298.548828125
x loss: 0.32502437631289166
y loss: 0.3996403714021047
w loss: 1139.4257507324219
h loss: 1012.723134358724
conf loss: 15.961354891459147
cls loss: 0.560105582078298
recall: 0.0
precision: 0.0
************epoch: 27 ***************
total loss: 4103.248046875
loss1:245.30926513671875, loss2: 1389.0126953125, loss3: 2468.926025390625
x loss: 0.29152263204256695
y loss: 0.2758776346842448
w loss: 534.8623021443685
h loss: 815.5395711263021
conf loss: 16.2246831258138
cls loss: 0.5553083817164103
recall: 0.0
precision: 0.0
************epoch: 28 ***************
total loss: 2999.522705078125
loss1:336.9549255371094, loss2: 1752.310302734375, loss3: 910.2576293945312
x loss: 0.2644174098968506
y loss: 0.44414860010147095
w loss: 334.78647867838544
h loss: 648.9955800374349
conf loss: 14.817063967386881
cls loss: 0.5332703590393066
recall: 0.0
precision: 0.0
************epoch: 29 ***************
total loss: 3979.97607421875
loss1:374.10736083984375, loss2: 1859.2445068359375, loss3: 1746.6241455078125
x loss: 0.27098152538140613
y loss: 0.29165806372960407
w loss: 537.3419036865234
h loss: 772.4309895833334
conf loss: 15.769588788350424
cls loss: 0.5535270174344381
recall: 0.0
precision: 0.0
Save Model at ./models/yolo/30.pth

************epoch: 30 ***************
total loss: 3949.78466796875
loss1:290.5174865722656, loss2: 1260.368896484375, loss3: 2398.898193359375
x loss: 0.31959329048792523
y loss: 0.3391527831554413
w loss: 684.0392379760742
h loss: 616.7203674316406
conf loss: 14.607483545939127
cls loss: 0.5690003037452698
recall: 0.0
precision: 0.0
************epoch: 31 ***************
total loss: 1981.303466796875
loss1:355.115478515625, loss2: 776.53955078125, loss3: 849.6483764648438
x loss: 0.33924222985903424
y loss: 0.32605303327242535
w loss: 366.0008544921875
h loss: 278.5222473144531
conf loss: 14.699036280314127
cls loss: 0.5470494826634725
recall: 0.0
precision: 0.0
************epoch: 32 ***************
total loss: 2188.74462890625
loss1:166.44534301757812, loss2: 1283.809814453125, loss3: 738.489501953125
x loss: 0.3520261347293854
y loss: 0.3979433973630269
w loss: 274.4340311686198
h loss: 438.67093658447266
conf loss: 15.160051981608072
cls loss: 0.5665518244107565
recall: 0.0
precision: 0.0
************epoch: 33 ***************
total loss: 2202.96533203125
loss1:191.62582397460938, loss2: 998.6744995117188, loss3: 1012.6649169921875
x loss: 0.3084101974964142
y loss: 0.2975728412469228
w loss: 300.4658025105794
h loss: 418.7908070882161
conf loss: 13.889191309611002
cls loss: 0.569959302743276
recall: 0.0
precision: 0.0
************epoch: 34 ***************
total loss: 2302.44921875
loss1:200.84307861328125, loss2: 1039.15966796875, loss3: 1062.4464111328125
x loss: 0.36394720276196796
y loss: 0.37850836912790936
w loss: 312.16450754801434
h loss: 440.73814900716144
conf loss: 13.293774604797363
cls loss: 0.5441252390543619
recall: 0.0
precision: 0.0
************epoch: 35 ***************
total loss: 4405.984375
loss1:284.1207275390625, loss2: 1016.5211181640625, loss3: 3105.3427734375
x loss: 0.3626277446746826
y loss: 0.2699126899242401
w loss: 693.1200205485026
h loss: 760.0078659057617
conf loss: 14.338323911031088
cls loss: 0.5628648400306702
recall: 0.0
precision: 0.0
************epoch: 36 ***************
total loss: 1913.2469482421875
loss1:211.41213989257812, loss2: 536.5771484375, loss3: 1165.2576904296875
x loss: 0.39507757623990375
y loss: 0.2833245297273
w loss: 348.30287679036456
h loss: 273.7943623860677
conf loss: 14.416383425394693
cls loss: 0.5569764673709869
recall: 0.0
precision: 0.0
************epoch: 37 ***************
total loss: 1334.109375
loss1:98.77485656738281, loss2: 627.5885009765625, loss3: 607.7459716796875
x loss: 0.31386294960975647
y loss: 0.31856351097424823
w loss: 217.62845166524252
h loss: 212.49593353271484
conf loss: 13.38981548945109
cls loss: 0.5564960936705271
recall: 0.0
precision: 0.0
************epoch: 38 ***************
total loss: 2304.2548828125
loss1:180.39263916015625, loss2: 998.72412109375, loss3: 1125.1380615234375
x loss: 0.3142807682355245
y loss: 0.3134910662968953
w loss: 362.3257548014323
h loss: 390.6422576904297
conf loss: 13.91358757019043
cls loss: 0.5755724112192789
recall: 0.0
precision: 0.0

```

**Comment by HzZHoO** [comment]
Did you solve this, I met the same problem..
